### PR TITLE
fix(@angular-devkit/build-angular): jasmine.clock with app builder

### PR DESCRIPTION
**Pull request by jkrems**
Ports the logic from WTR and webpack and adds a regression test for this behavior.

Fixes https://github.com/angular/angular-cli/issues/28813

### CODE REVIEW

**Comment by jkrems** [comment]
Somehow this broke existing behavior. Converting to draft for now while I investigate.

**Review by clydin — APPROVED** [review]
Nice work.
just one typo otherwise LGTM.

File: packages/angular_devkit/build_angular/src/builders/karma/application_builder.ts [hunk 1]
```diff
@@ -106,6 +106,66 @@ class AngularAssetsMiddleware {
...
+    return {
+      // This has to be a "reporter" because reporters run _after_ frameworks
+      // and karma-jasmine-html-reporter injects additional scripts that may
+      // depend on Jasmine but aren't modules - which means that they would run
+      // _before_ all module code (including jasmine).
+      [`reporter:${AngularPolyfillsPlugin.NAME}`]: [
+        'factory',
+        Object.assign((files: (string | FilePattern)[]) => {
+          // The correct order is zone.js -> jasmine -> zone.js/testing.
+          // Jasmine has to see the patched version of the global `setTimeou`t
```
[the remark this answers is not in the document]
Comment (jkrems) [reply packages/angular_devkit/build_angular/src/builders/karma/application_builder.ts]: Thanks!

**Comment by jkrems** [comment]
The changes were merged into the following branches: main, 19.0.x

**Pull request merged**